core/lib/types/Nat.agda

```diff
@@ -77,6 +78,13 @@ instance
   ℕ-level {⟨-2⟩} = ℕ-is-set
   ℕ-level {n = S n} = raise-level (S (S n)) ℕ-level
 
+even : ℕ → Bool
+even O = true
+even (S n) = negate (even n)
+
+odd : ℕ → Bool
+odd = negate ∘ even
+
```

Comment:
@timjb I wonder `even (S n) = odd n` and `odd (S n) = even n` would be easier because of the double negation in the following code. I could be wrong.

Reply:
@favonia One advantage of the definition `odd = negate ∘ even` is that `odd n`, `even n`, `odd (S n)`, `even (S n)`, ... all reduce to `even n`. With `even` and `odd` defined by mutual recursion, there are two propositions `even n` and `odd n` for every `n` that are not judgmentally related. So you would have to prove (by induction) that e.g. `odd n == negate (even n)` and `even n == negate (odd n)`. With the current definitions of `even` and `odd`, the first path is reflexivity and the second path is an instance of `!-!`.